
<!--Libraries-->
```{r libs, echo=FALSE, warning=FALSE}
library(MutationalPatterns)
library(BSgenome)
library(GenomicRanges)
library(ggpubr)
library(reshape)
library(pheatmap)
library(gdata)
```

## Analysis

Load and parse data, match IDs:

```{r readdat, echo=FALSE}

# Load mutation data:
load("snvs.RData")

# Load HN data:
load("dat.fullannot.plusclon.2021.RData")

# Keep only snvs for samples with Nanostring data:
snvs <- snvs[which(snvs$SampleID %in% dat.fullannot$Sample),]

```

### Mut signature analysis

```{r mutSigs}

ref_genome <- "BSgenome.Hsapiens.UCSC.hg38"
library(ref_genome, character.only = TRUE)

# vcf_files <- list.files(system.file("extdata", package="MutationalPatterns"),
# pattern = ".vcf", full.names = TRUE)
# sample_names <- c(
# "colon1", "colon2", "colon3",
# "intestine1", "intestine2", "intestine3",
# "liver1", "liver2", "liver3")
# vcfs <- read_vcfs_as_granges(vcf_files, sample_names, ref_genome)

snvs$ALT <- snvs$Alt
snvs$REF <- snvs$Ref
snvs$End <- snvs$Start
# select only point mutations:
snvs.pm <- snvs[which((snvs$REF %in% c("A","C","G","T"))&
                        (snvs$ALT %in% c("A","C","G","T"))),]
vcfs <- list()
i<-0
for (sample in unique(snvs.pm$SampleID)) {
  i <- i+1
  currentsample <- snvs.pm[which(snvs.pm$SampleID == sample),]
  vcfs[[i]] <- makeGRangesFromDataFrame(currentsample[,c("SampleID",
                                         "Chr","Start",
                                         "End",
                                         "REF","ALT")],
                                 keep.extra.columns=TRUE)
}
names(vcfs) <- unique(snvs.pm$SampleID)

# #muts = mutations_from_vcf(vcfs)
# #types = mut_type(vcfs)
# #context = mut_context(vcfs, ref_genome)
# #type_context = type_context(vcfs, ref_genome)
# # type_occurrences <- mut_type_occurrences(vcfs, ref_genome)
# # save(type_occurrences, file="MSIG.type_occurrences.VARDICTupdated20200817.RData")
# 
# load("../scripts_forpub/MSIG.type_occurrences.VARDICTupdated20200817.RData")
# 
# ## metadata on immune group:
# commonSamples <- intersect(trimws(as.character(unique(snvs$SampleID))),trimws(as.character(dat.fullannot$Sample)))
# # immgr <- NULL
# # for (sample in commonSamples) {
# #   immgr <- c(immgr,
# #                dat.fullannot[which(dat.fullannot$Sample == sample),]$ImmunityGroup)
# # }
# 
# tpocc <- type_occurrences[which(rownames(type_occurrences) %in% commonSamples),]
# immgr <- dat.fullannot[which(dat.fullannot$Sample %in% tpocc),]$ImmunityGroup
# 
# 
# pdf("plots2021/mutsig/mutationalSpectrum.pdf")
# plot_spectrum(type_occurrences[which(rownames(type_occurrences) %in% commonSamples),], 
#               CT = TRUE, by = immgr,
#               legend = TRUE)
# dev.off()
# 
# # Calculate TMB and add to data frame:
# 
# ### Does TMB associate with immunity group?
# 
# # pdf("plots/TMBvsGroup.pdf")
# # ggplot(dat.hn[which(dat.hn$TMB_missense < 2000),], aes(TMB_missense, EOMES))+
# #   geom_point()+
# #   geom_smooth(method=lm)
# # dev.off()
# 
# my_comparisons <- list(c("Absent","MacrophageRich"),
#                        c("MacrophageRich","DCrich"),
#                        c("DCrich","PCrich"),
#                        c("PCrich","Cytotoxic"))



```

Mutational signature analysis using deconstructSigs:

```{r}
library(deconstructSigs)
library("BSgenome.Hsapiens.UCSC.hg38")

# Build input for deconstruction (all samples and SNVs together):
sigs.input <- mut.to.sigs.input(mut.ref = snvs.pm, 
                                sample.id = "SampleID", 
                                chr = "Chr", 
                                pos = "Start", 
                                ref = "REF", 
                                alt = "ALT",
                                bsg = BSgenome.Hsapiens.UCSC.hg38)

# Load reference signatures:
load("~/Desktop/UCL/projects/deconstructSigs/data/signatures.exome.cosmic.v3.may2019.rda")

# Run deconstructSigs one sample at a time and save everything into the sigs data frame:
sigs <- NULL
i <- 0
for (sample in rownames(sigs.input)) {
  i <- i+1
  print(i)
  sigs_1 = whichSignatures(tumor.ref = sigs.input, 
                             signatures.ref = signatures.exome.cosmic.v3.may2019,
                             sample.id = sample, 
                             contexts.needed = TRUE,
                             signature.cutoff = 0,
                             tri.counts.method = 'default')
  sigs <- rbind(sigs,sigs_1$weights)
  
}
sigs <- data.frame(sigs)

# Remove artefactual sigs:
artefacts <- paste0("SBS",c(27,43,45:60))
sigs <- sigs[,setdiff(colnames(sigs),artefacts)]
sigs$SampleID <- rownames(sigs)

df.sigs <- melt(sigs)

pdf("plots2021/mutsig/deconstructsigs/boxplot.pdf",w=6,h=20)
ggplot(df.sigs, aes(variable,value))+
  geom_boxplot()+
  coord_flip()
dev.off()

# Check which sigs have a contribution of at least 5% in some samples:
# average contribution:
selsig <- names(rev(sort(apply(sigs[,1:47],2,function(x) length(which(x>0.05)))))[1:11])

#selsig <- names(which(apply(sigs[,1:47],2,function(x) mean(x))>0.01))


colnames(df.sigs) <- c("Sample","Signature","Exposure")
df.sigs.merged.decon <- merge(df.sigs,
            dat.fullannot[,c("Sample","ImmunityGroup","TMBtotal","Smoking.Category","HPV.category")],
            all.x=FALSE, all.y=FALSE,
            by.x="Sample", by.y="Sample") 

pdf("plots2021/mutsig/deconstructsigs/signatureContribByGroup.pdf",w=25,h=25)
ggboxplot(df.sigs.merged.decon, 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7","#FFE89F",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("Immunity group")+
  ylab("Total mutations contributed (log 10)")+
  facet_wrap(~Signature,scales = "free")+
  theme(axis.text.x=element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 0)
dev.off()

pdf("plots2021/mutsig/deconstructsigs/signatureContribByGroupselected.pdf",w=12,h=8)
ggboxplot(df.sigs.merged.decon[which(df.sigs.merged.decon$Signature %in% selsig),], 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7","#FFE89F",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("Immunity group")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free",nrow=2)+
  theme(axis.text.x=element_blank(),
        axis.ticks.x = element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()

pdf("plots2021/mutsig/deconstructsigs/signatureContribByGroupselectedS2S13.pdf",w=6,h=6)
ggboxplot(df.sigs.merged.decon[which(df.sigs.merged.decon$Signature %in% c("SBS2","SBS13")),], 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7","#FFE89F",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("Immunity group")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free")+
  theme(axis.text.x=element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()


save(df.sigs.merged.decon, file="df.sigs.merged.decon.exome.updated2021.RData")

# test by HPV status:
t.test(df.sigs.merged.decon[which((df.sigs.merged.decon$HPV.category == "Positive") & (df.sigs.merged.decon$Signature == "SBS2")),]$Exposure,
df.sigs.merged.decon[which((df.sigs.merged.decon$HPV.category == "Negative") & (df.sigs.merged.decon$Signature == "SBS2")),]$Exposure,alternative = "greater")

my_comparisonssmoke <- list(c("Present","Past"), c("Past","Never"),c("Present","Never"))
pdf("plots2021/mutsig/deconstructSigs/signatureContribBySmoking.onlySBS4.pdf")
ggboxplot(df.sigs.merged.decon[which(df.sigs.merged.decon$Signature=="SBS4",
                           (!is.na(df.sigs.merged.decon$Smoking.Category))),], 
          x = "Smoking.Category", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "Smoking.Category")+
  xlab("Smoking status")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free",nrow=2)+
  stat_compare_means(comparisons = my_comparisonssmoke, label = "p.signif")+
  stat_compare_means(label.y = 0)
dev.off()



```

Mutational signature analysis:
```{r mutsiganalysis}
### building mutational matrix:
mut_mat <- mut_matrix(vcf_list = vcfs, 
                      ref_genome = ref_genome)
save(mut_mat, file="MSIG.mut_matVARDICTnew.updated20200817.RData")

# First add a small psuedocount to your mutation count matrix:
mut_mat <- mut_mat + 0.0001

#Use the NMF package to generate an estimate rank plot:
library("NMF")
estimate <- nmf(mut_mat, rank=3:15, method="brunet", nrun=10, seed=123456)
#And plot it:
pdf("plots2020.updated/mutsig/rankEstimate.pdf")
plot(estimate)
dev.off()

## Choosing rank 8 (i.e. 8 sigs) based on the stats:
nmf_res <- extract_signatures(mut_mat, rank = 5, nrun = 1000)
save(nmf_res,file="MSIG.nmf_res.VARDICTnew5.updated20200817.RData")

# Assign signature names:
colnames(nmf_res$signatures) <- paste0("S",1:5)
rownames(nmf_res$contribution) <- paste0("S",1:5)

# Plot the 96-profile of the signatures:
pdf("plots2020.updated/mutsig/mutsig5/signatureProfiles.pdf")
plot_96_profile(nmf_res$signatures, condensed = TRUE)
dev.off()


#Download mutational signatures from the COSMIC website:
cancer_signatures = read.csv("sigProfiler_exome_SBS_signatures.csv")
# Match the order of the mutation types to MutationalPatterns standard
cancer_signatures$Somatic.Mutation.Type <-
  apply(cancer_signatures[,1:2],1,
        function(x) paste0(substr(x[2],1,1),"[",x[1],"]",
                           substr(x[2],3,3)))
new_order = match(row.names(mut_mat), cancer_signatures$Somatic.Mutation.Type)
# Reorder cancer signatures dataframe
cancer_signatures = cancer_signatures[as.vector(new_order),]
# Add trinucletiode changes names as row.names
row.names(cancer_signatures) = cancer_signatures$Somatic.Mutation.Type
# Keep only 96 contributions of the signatures in matrix
cancer_signatures = as.matrix(cancer_signatures[,-c(1,2,ncol(cancer_signatures))])

#Calculate pairwise cosine similarity between mutational profiles and COSMIC signatures:
cos_sim_samples_signatures = cos_sim_matrix(mut_mat, cancer_signatures)

 hclust_cosmic = cluster_signatures(cancer_signatures, method = "average")
# store signatures in new order
cosmic_order = colnames(cancer_signatures)[hclust_cosmic$order]
 
# Plot heatmap with specified signature order
pdf("plots2020.updated/mutsig/mutsig5/cosmicSigprevalence.pdf",w=20)
plot_cosine_heatmap(cos_sim_samples_signatures,
 col_order = cosmic_order,
 cluster_rows = TRUE)
dev.off()

# Cosine similarities
cos_sim_samples_signatures = cos_sim_matrix(nmf_res$signatures, cancer_signatures)
pdf("plots2020.updated/mutsig/mutsig5/cosineSimilaritiesCOSMIC2.pdf",w=15)
plot_cosine_heatmap(cos_sim_samples_signatures,
 cluster_rows = TRUE)
dev.off()

assignedSigs <- NULL
for (s in 1:5) {
  print(paste(s,":"))
  c <- cos_sim_samples_signatures[s,][which(cos_sim_samples_signatures[s,]==max(cos_sim_samples_signatures[s,]))]
  print(c)
  assignedSigs <- c(assignedSigs,names(c))
}

# total number of mutations per siganture
        total_signatures = colSums(nmf_res$signatures) 

        # calculate signature contribution in absolute number of signatures
        abs_contribution = nmf_res$contribution * total_signatures

# Matrix with signature prevalence:
nmf_sigs <- abs_contribution
# Rename rows according to assigned signatures:
rownames(nmf_sigs) <- assignedSigs

nmf_sigs <- nmf_sigs[which(!(rownames(nmf_sigs) %in% artefacts)),]

save(nmf_sigs, file="nmf_sigs_5_exome.RData")
        
df.sigs <- melt(t(nmf_sigs))  
colnames(df.sigs) <- c("Sample","Signature","Exposure")

df.sigs.merged <- merge(df.sigs,
            dat.fullannot[,c("Sample","ImmunityGroup","TMBtotal","Smoker")],
            all.x=FALSE, all.y=FALSE,
            by.x="Sample", by.y="Sample") 

save(df.sigs.merged, file="df.sigs.merged5_exome.RData")    

#### Compare deconstructSigs and MutPat sigs:
load("nmf_sigs_7.RData")
rel_contribution <- nmf_sigs
for (colN in 1:ncol(rel_contribution)) {
  rel_contribution[,colN] <- rel_contribution[,colN]/colSums(rel_contribution)[colN]
}  
# Matrix with signature prevalence:
nmf_sigs <- rel_contribution

sigs.decons.subset <- as.data.frame(sigs[,rownames(nmf_sigs)])
sigs.decons.subset$SampleID <- rownames(sigs.decons.subset)

df.decon <- melt(sigs.decons.subset)
colnames(df.decon) <- c("SampleID","Signature","DeconstructSigs")
nmf5 <- as.data.frame(t(nmf_sigs))
nmf5$SampleID <- rownames(nmf5)
df.nmf5 <- melt(nmf5)
colnames(df.nmf5) <- c("SampleID","Signature","NMF5")

df.both <- merge(df.decon, df.nmf5,
                 by.x=c("SampleID","Signature"),
                 by.y=c("SampleID","Signature"),
                 all.x=FALSE, all.y=FALSE)

pdf("plots2020.updated/mutsig/compareDeconNMF5.pdf")
ggplot(df.both,aes(DeconstructSigs,NMF5))+
  geom_point()+
  geom_smooth(method=lm)+
  facet_wrap(~Signature,scale="free")
dev.off()

for (s in rownames(nmf_sigs)) {
  print(paste0("Signature",s,":"))
  print(cor.test(nmf_sigs[s,],sigs[,s]))
}
cor.test(nmf_sigs["SBS2",],sigs[,"SBS2"])
         
         
df.sigs.merged$log10Exp <- log10(df.sigs.merged$Exposure)

#load("df.sigs.merged6.RData")

pdf("plots2020.updated/mutsig/mutsig5/signatureContribByGroup.pdf",w=15,h=6)
ggboxplot(df.sigs.merged, 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("Immunity group")+
  ylab("Total mutations contributed (log 10)")+
  facet_wrap(~Signature,scales = "free",nrow=1)+
  theme(axis.text.x=element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()


# ##### What about relative signatures?
# rel_contribution <- nmf_sigs
# for (colN in 1:ncol(rel_contribution)) {
#   rel_contribution[,colN] <- rel_contribution[,colN]/colSums(rel_contribution)[colN]
# }  

# Matrix with signature prevalence:
#nmf_sigs <- rel_contribution
        
df.sigs <- melt(t(nmf_sigs))  
colnames(df.sigs) <- c("Sample","Signature","Exposure")
df.sigs.merged <- merge(df.sigs,
            dat.fullannot[,c("Sample","ImmunityGroup","TMBtotal","Smoker","HPV.Status")],
            all.x=FALSE, all.y=FALSE,
            by.x="Sample", by.y="Sample") 

save(df.sigs.merged, file="df.sigs.merged.relative5_exome.RData")  

pdf("plots2020.updated/mutsig/mutsig5/relativeSignatureContribByGroup.pdf",w=15,h=6)
ggboxplot(df.sigs.merged, 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("Immunity group")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free",nrow=1)+
  theme(axis.text.x=element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()

pdf("plots2020.updated/mutsig/mutsig6.chosen/relativeSignatureContrib.selected.2.pdf",w=6,h=6)
ggboxplot(df.sigs.merged[which(df.sigs.merged$Signature %in% c("SBS2","SBS4")),], 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("")+
  ylab("Exposure")+
  facet_wrap(~Signature,nrow=1)+
  theme(axis.text.x=element_blank(),
        axis.ticks.x = element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()

 my_comparisonssmoke <- list(c("Present","Past"), c("Past","Never"),c("Present","Never"))
pdf("plots2020.updated/mutsig/mutsig6.chosen/signatureContribBySmoking.S4.2.pdf",w=4,h=4)
ggboxplot(df.sigs.merged[which((!is.na(df.sigs.merged$Smoker))&(df.sigs.merged$Signature=="SBS4")),],
          x = "Smoker",
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "Smoker")+
  xlab("Smoking status")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free",nrow=1)+
  theme(axis.text.x=element_blank())+
  stat_compare_means(comparisons = my_comparisonssmoke, label = "p.signif")+
  stat_compare_means(label.y = 0)
dev.off()


```

## COSMIC signatures:
```{r}
fit_res <- fit_to_signatures(mut_mat, cancer_signatures)
# Select signatures with some contribution
#select <- names(which(rowSums(fit_res$contribution) >410))
# Plot contribution barplot
plot_contribution(fit_res$contribution[select,],
 cancer_signatures[,select],
 coord_flip = FALSE,
 mode = "absolute")

save(fit_res, file="fit_res_cosmic.updated20200817.RData")

pdf("plots2020.updated/mutsig/cosmic/signatureProfiles_patternPlot.pdf",w=7,h=3)
plot_96_profile(cancer_signatures[,c("SBS2","SBS13","SBS4")], 
                condensed = TRUE)
dev.off()

pdf("plots2020.updated/mutsig/cosmic/COSMICrelativeContributions.pdf")
plot_contribution_heatmap(fit_res$contribution[selectedSigs,],
 cluster_samples = TRUE,
 method = "complete")  
dev.off()

rel_contribution <- fit_res$contribution
for (colN in 1:ncol(rel_contribution)) {
  rel_contribution[,colN] <- rel_contribution[,colN]/colSums(rel_contribution)[colN]
}  

select <- names(rev(sort(apply(rel_contribution,1,function(x) length(which(x>0.05)))))[1:10])

selectedSigs <- setdiff(select,artefacts)

df.cosmic <- melt(t(rel_contribution[selectedSigs,]))
colnames(df.cosmic) <- c("Sample","Signature","Exposure")
df.cosmic.merged <- merge(df.cosmic,
            dat.fullannot[,c("Sample","ImmunityGroup","TMB","Smoker")],
            all.x=FALSE, all.y=FALSE,
            by.x="Sample", by.y="Sample") 

save(df.cosmic.merged, file="df.cosmic.merged.new_exome.RData")  

pdf("plots2020.updated/mutsig/cosmic/relativeSignatureContribByGroup.pdf",w=12,h=14)
ggboxplot(df.cosmic.merged, 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("Immunity group")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free",nrow=4)+
  theme(axis.text.x=element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()

pdf("plots2020.updated/mutsig/cosmic/relativeSignatureContribByGroup.selectedSet.pdf",w=9,h=11)
ggboxplot(df.cosmic.merged[which(df.cosmic.merged$Signature
                                 %in% selectedSigs),], 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("Immunity group")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free",nrow=3)+
  theme(axis.text.x=element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()

pdf("plots2020.updated/mutsig/cosmic/relativeSignatureContribByGroup.selected.pdf",w=6)
ggboxplot(df.cosmic.merged[which(df.cosmic.merged$Signature
                                 %in% c("SBS2","SBS13","SBS44")),], 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("Immunity group")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free")+
  theme(axis.text.x=element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()

pdf("plots2020.updated/mutsig/cosmic/relativeSignatureContribByGroup.selectedS2134.pdf",w=6,h=6)
ggboxplot(df.cosmic.merged[which(df.cosmic.merged$Signature
                                 %in% c("SBS2","SBS13","SBS4")),], 
          x = "ImmunityGroup", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "ImmunityGroup")+
  xlab("Immunity group")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free")+
  theme(axis.text.x=element_blank())+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()

my_comparisonssmoke <- list(c("Present","Past"), c("Past","Never"),c("Present","Never"))
pdf("plots2020.updated/mutsig/cosmic/signatureContribBySmoking.onlySBS4.pdf")
ggboxplot(df.cosmic.merged[which(df.cosmic.merged$Signature=="SBS4",
                           (!is.na(df.cosmic.merged$Smoker))),], 
          x = "Smoker", 
          y = "Exposure",
          palette = c("#4A6FA5", "#DCD6F7",
                      "#F98948","#993955"),
          fill = "Smoker")+
  xlab("Smoking status")+
  ylab("Exposure")+
  facet_wrap(~Signature,scales = "free",nrow=2)+
  stat_compare_means(comparisons = my_comparisonssmoke, label = "p.signif")+
  stat_compare_means(label.y = 0)
dev.off()

save(df.cosmic, file="df.cosmic.new_exome.updated20200817.RData")
save(df.cosmic.merged, file="df.cosmic.merged.new_exome.updated20200817.RData")

# Compare COSMIC and deconstructSigs:
df.decon <- melt(sigs)
colnames(df.decon) <- c("SampleID","Signature","DeconstructSigs")
cosmic <- as.data.frame(t(nmf_sigs))
colnames(df.cosmic)[3] <- "COSMIC"

df.both <- merge(df.decon, df.cosmic,
                 by.x=c("SampleID","Signature"),
                 by.y=c("Sample","Signature"),
                 all.x=FALSE, all.y=FALSE)

pdf("plots2020.updated/mutsig/compareDeconCOSMIC.withNMF.pdf",w=8,h=8)
ggplot(df.both,aes(DeconstructSigs,COSMIC))+
  geom_point()+
  geom_smooth(method=lm)+
  facet_wrap(~Signature,scale="free")
dev.off()

for (s in unique(df.both$Signature)) {
  print(paste0("Signature",s,":"))
  print(cor.test(df.both[which(df.both$Signature==s),]$DeconstructSigs,df.both[which(df.both$Signature==s),]$COSMIC))
}

```


Finally, correlate APOBEC expression and BER expression with signatures, and check for specific differences between groups:
```{r}

dat.hn <- read.table("../HN_MASTERCOHORT_withExhGroups11-15-2017.txt", header=TRUE,
                     sep="\t", stringsAsFactors = FALSE)

dat.merged <- merge(dat.hn, 
                    dat.annot.plusimm,
                    all.x=FALSE, all.y=FALSE,
                    by.x="Sample.ID", by.y="Sample")

## no APOBEC genes

## Try BER genes:
ddr <- read.xls("~/Desktop/UCL/supervisedProjects/DanielJacobson/Rotation1-master/DDRpathways.xlsx")
ber <- ddr[which(ddr$Pathway.1 == "BER"),]$Gene.ID

which(colnames(dat.merged) %in% ber)
colnames(dat.merged)[437]
# a single gene HMGB1 so not worth testing further

```


Further tests on SBS2:
```{r}

#load("df.cosmic.merged.new_exome.RData")
load("dat.fullannot.RData")
df.infoplussig <- merge(df.sigs,
                        dat.fullannot[,c("Sample","TMBtotal",
                                         "HPV.Status")],
                        by.x="Sample",by.y="Sample",
                        all.x=FALSE,all.y=FALSE)

sbs2 <- df.infoplussig[which(df.infoplussig$Signature == "SBS2"),]
sbs13 <- df.infoplussig[which(df.infoplussig$Signature == "SBS13"),]

wilcox.test(sbs2[which(sbs2$HPV.Status=="Positive"),]$Exposure,
       sbs2[which(sbs2$HPV.Status=="Negative"),]$Exposure,
       alternative = "greater")
t.test(sbs13[which(sbs30$HPV.Status=="Positive"),]$Exposure,
       sbs13[which(sbs30$HPV.Status=="Negative"),]$Exposure,
       alternative = "greater")

cor.test(sbs2$Exposure,sbs2$TMBtotal)
cor.test(sbs13$Exposure,sbs13$TMBtotal)


```

