---
title: "WES data processing and integration"
author: "Maria Secrier"
date: "25/04/2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Description

This script parses the NGS data in the H&N cohort. 

<!--Libraries-->
```{r libs, echo=FALSE, warning=FALSE}
library(gdata)
library(pheatmap)
library(reshape)
library(ggplot2)
library(GenVisR)
library(survival)
library(RColorBrewer)
library(ComplexHeatmap)
library(GSVA)
library(pheatmap)
library(ggpubr)
```

## Analysis

Load and parse data, match IDs:

```{r readdat, echo=FALSE}

# Load HN data:
load("dat.annot.plusimm.2021.RData")
dat.annot.plusimm$ImmunityGroup <- factor(dat.annot.plusimm$ImmunityGroup,
                      levels=c("Absent","MacrophageRich","DCrich","PCrich","Cytotoxic"))


# Load mutation data:
snvs <- read.delim("../scripts_forpub/vardict.pass_20200617.tsv")
table(snvs$Functional_Class)
hist(snvs$AlleleFreq)
quantile(snvs$AlleleFreq, na.rm=TRUE)

# Read sample info:
sample.info <- read.xls("../scchn_wes_nanostring.xlsx")
snvs <- merge(snvs, sample.info[,c("SampleID","SampleName")],
              all.x=FALSE, all.y = FALSE,
              by.x="Sample", by.y="SampleName")
save(snvs, file="snvs.RData")

snvs.nonsyn <- snvs[which(snvs$Functional_Class %in%
                            c("missense_variant",
                              "frameshift_variant",
                              "stop_gained",
                              "5_prime_UTR_premature_start_codon_gain_variant",
                              "disruptive_inframe_deletion",
                              "disruptive_inframe_insertion",
                              "start_lost",
                              "stop_lost","gene_fusion",
                              "bidirectional_gene_fusion")),]
snvs.missense <- snvs[which(snvs$Functional_Class ==
                            "missense_variant"),]

```

Generic stats:
```{r}

mt <- table(snvs$SampleID)
tmb.total <- data.frame(mt)
colnames(tmb.total) <- c("SampleID","TMBtotal")
tmb.nonsyn <- data.frame(table(snvs.nonsyn$SampleID))
colnames(tmb.nonsyn) <- c("SampleID","TMBnonsyn")
tmb.missense <- data.frame(table(snvs.missense$SampleID))
colnames(tmb.missense) <- c("SampleID","TMBmissense")

dat.fullannot <- merge(dat.annot.plusimm,
                       tmb.total,
                       by.x="Sample",by.y="SampleID",
                       all.x=FALSE,all.y=FALSE)
dat.fullannot <- merge(dat.fullannot,
                       tmb.nonsyn,
                       by.x="Sample",by.y="SampleID",
                       all.x=TRUE,all.y=FALSE)
dat.fullannot <- merge(dat.fullannot,
                       tmb.missense,
                       by.x="Sample",by.y="SampleID",
                       all.x=TRUE,all.y=FALSE)

min(dat.fullannot$TMBtotal)
max(dat.fullannot$TMBtotal)
median(dat.fullannot$TMBtotal)
dat.fullannot$MutRate <- dat.fullannot$TMBtotal/95505476*10^6
  #88341206*10^6 #divided by exome
 #length from UCSC table browser

dat.fullannot$log10TMBtot <- log10(dat.fullannot$TMBtotal+1)
dat.fullannot$log10TMBnonsyn <- log10(dat.fullannot$TMBnonsyn+1)
dat.fullannot$log10TMBmissense <- log10(dat.fullannot$TMBmissense+1)

# Merge with clinical data:
clin <- read.xls("../ts_io_head_neck_clinical_data.xlsx")
dat.fullannot <- merge(dat.fullannot, clin[,c("Sample.ID","CD8.Ct.Cells.Mm2","CD8.Im.Cells.Mm2",
                                       "CD3.Ct.Cells.Mm2","CD3.Im.Cells.Mm2",
                                       "Density.FOXP3.TC","Density.FOXP3.IM",
                                       "Density.GITR.TC","Density.GITR.IM",
                                       "ATM.Positive.10.Percent","ATM.Total.Percentage",
                                         "CCNE1.Nuclear.H.Score","CCNE1.Total.Percentage",
                                       "c.myc.cytoplasmic.H.Score","C.Myc.Total.Cytoplasmic",
                                         "LKB1.cytoplasmic.H.Score","LKB1.Total.Cytoplasmic",
                                       "PTEN.cytoplasmic.H.Score","PTEN.Cytoplasmic.Total",
                                       "Age","Age.Category","TMB","Packs.per.Year","Packs.per.Day",
                                       "Overall.Survival.months.","Censoring.Status",
                                       "PFS","Censoring.Status..PFS.")],
                    by.x="Sample", by.y="Sample.ID",
                    all.x=TRUE, all.y=FALSE)
save(dat.fullannot, file="dat.fullannot.20210726.RData")


### Now check correlations with various parameters:
pdf("plots2021/corr.TMBvsProdClon.pdf")
ggscatter(dat.fullannot,#[which(dat.fullannot$TMBtotal<10000),],
          x = "log10TMBtot", y = "Productive.Clonality",
          add = "reg.line", conf.int = TRUE,
          cor.coef = TRUE) 
dev.off()
pdf("plots2021/corr.TMBvsObsRich.pdf")
ggscatter(dat.fullannot,#[which(dat.fullannot$TMBtotal<10000),],
          x = "log10TMBtot", y = "observed_richness",
          add = "reg.line", conf.int = TRUE,
          cor.coef = TRUE) 
dev.off()
pdf("plots2021/corr.TMBvsPDL1.pdf")
ggscatter(dat.fullannot,#[which(dat.fullannot$TMBtotal<10000),],
          x = "log10TMBtot", y = "PD.L1.SP263.TC",
          add = "reg.line", conf.int = TRUE,
          cor.coef = TRUE)
dev.off()
ggscatter(dat.fullannot,#[which(dat.fullannot$TMBnonsyn<10000),],
          x = "log10TMBnonsyn", y = "PDL1_SP263_PercTumCell.continuous",
          add = "reg.line", conf.int = TRUE,
          cor.coef = TRUE) 
ggscatter(dat.fullannot,#[which(dat.fullannot$TMBmissense<7000),],
          x = "log10TMBmissense", y = "PDL1_SP263_PercTumCell.continuous",
          add = "reg.line", conf.int = TRUE,
          cor.coef = TRUE) 
ggscatter(dat.fullannot,#[which(dat.fullannot$TMBmissense<7000),],
          x = "log10TMBmissense", y = "T_cells_CD4",
          add = "reg.line", conf.int = TRUE,
          cor.coef = TRUE) 
ggscatter(dat.fullannot[which(dat.fullannot$TMBnonsyn<7000),],
          x = "log10TMBnonsyn", y = "Cytotoxic_cells",
          add = "reg.line", conf.int = TRUE,
          cor.coef = TRUE) 
pdf("plots2021/TMBtot_vsCD8.pdf")
ggscatter(dat.fullannot,#[which(dat.fullannot$TMBmissense<7000),],
          x = "log10TMBtot", y = "CD8.Ct.Cells.Mm2",
          add = "reg.line", conf.int = TRUE,
          cor.coef = TRUE) 
dev.off()
ggscatter(dat.fullannot[which(dat.fullannot$TMBnonsyn<7000),],
          x = "log10TMBtot", y = "Cytotoxic_cells",
          add = "reg.line", conf.int = TRUE,
          cor.coef = TRUE) 

my_comparisons <- list(c("Absent","MacrophageRich"),
                       c("MacrophageRich","DCrich"),
                       c("DCrich","PCrich"),
                       c("PCrich","Cytotoxic"))

pdf("plots2021/TMBtotal_byimmg.pdf")
ggboxplot(dat.fullannot, x = "ImmunityGroup", 
         y = "log10TMBtot",
         palette = c("#4A6FA5", "#DCD6F7","#FFE89F",
                     "#F98948","#993955"),
         fill = "ImmunityGroup")+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  xlab("Immunity group")+
  ylab("Total mutational burden (log10)")+
  theme(axis.text.x=element_blank())+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()


pdf("plots2021/TMBnonsyn_byimmg.pdf")
ggboxplot(dat.fullannot, x = "ImmunityGroup", 
         y = "log10TMBnonsyn",
         palette = c("#4A6FA5", "#DCD6F7","#FFE89F",
                     "#F98948","#993955"),
         fill = "ImmunityGroup")+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  xlab("Immunity group")+
  ylab("Nonsynonymous mutational burden (log10)")+
  theme(axis.text.x=element_blank())+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()
pdf("plots2021/TMBmissense_byimmg.pdf")
ggboxplot(dat.fullannot, x = "ImmunityGroup", 
         y = "log10TMBmissense",
         palette = c("#4A6FA5", "#DCD6F7","#FFE89F",
                     "#F98948","#993955"),
         fill = "ImmunityGroup")+
  stat_compare_means(comparisons = my_comparisons, label = "p.signif")+
  xlab("Immunity group")+
  ylab("Missense mutational burden (log10)")+
  theme(axis.text.x=element_blank())+
  stat_compare_means(label.y = 0, label.x= 2)
dev.off()

```

Top mutated genes:

```{r}
### Plot top frequently mutated genes:

### First, keep only mutations of consequence:
#snvs.nonsyn <- snvs.nonsyn[which(snvs.nonsyn$Significance != "unknown"),]


geneFreq <- rev(sort(table(unique(snvs.nonsyn[,c("SampleID","Gene")])$Gene)))

# read in cancer genes:
cancerGenes <- read.table("Census_allWed Jul 21 12_38_18 2021.tsv", header=FALSE, stringsAsFactors = FALSE,sep="\t")$V1

# read in cancer genes:
#rtks <- read.table("../../../data/RTKgeneList.txt",
#                   header=TRUE, sep="\t",
#                   stringsAsFactors = FALSE)
#cancerGenes <- rtks$Gene

geneFreq.cancer <- rev(sort(geneFreq[which(names(geneFreq) %in% cancerGenes)]))

#snvs$MappedID <- sapply(snvs$Sample, function(x)
#  paste(strsplit(as.character(x),"_")[[1]],collapse="-"))

snvs.nonsyn.keep <- snvs.nonsyn[which((snvs.nonsyn$Gene %in% cancerGenes)&
  (snvs.nonsyn$SampleID %in% dat.fullannot$Sample) &
    (snvs.nonsyn$Effect != "LOW")),]


### Reformatting and renaming columns to comply with MAF annotation for the waterfall function:
snvs.MAF <- unique(snvs.nonsyn.keep[,c("SampleID","Gene","Functional_Class")])
colnames(snvs.MAF) <- c("sample","gene","variant_class")

genomic.MAF <- snvs.MAF

# specifying the variant class order:
vco.snv <- unique(snvs.MAF$variant_class)
vco <- unique(genomic.MAF$variant_class)

# oncoPrint(mat,
#     alter_fun = alter_fun, col = col, 
#     column_title = column_title, heatmap_legend_param = heatmap_legend_param)

```

### COMPLEX HEATMAPS:

```{r}

# First, need to restrict genomicMAF to top genes:
tb.count <- table(unique(genomic.MAF[,c("sample","gene")])$gene)
topgenes <- tb.count[which(tb.count>=10/100*94)]

genomic.MAF.topgenes <- genomic.MAF[which(genomic.MAF$gene %in% names(topgenes)),]

#### Trying to use complex heatmaps:

matMut <- array("",c(length(unique(genomic.MAF.topgenes$gene)),
                     length(unique(snvs.nonsyn.keep$SampleID))))
rownames(matMut) <- unique(genomic.MAF.topgenes$gene)
colnames(matMut) <- unique(snvs.nonsyn.keep$SampleID)
for (gene in unique(genomic.MAF.topgenes$gene)) {
  for (sample in unique(snvs.nonsyn.keep$SampleID)) {
    if (nrow(genomic.MAF.topgenes[which((genomic.MAF.topgenes$sample == sample)&
                               (genomic.MAF.topgenes$gene == gene)),])>0) {
      matMut[gene,sample] <- paste(unique(genomic.MAF.topgenes[which((genomic.MAF.topgenes$sample == sample)&
                      (genomic.MAF.topgenes$gene == gene)),]$variant_class),collapse=";")
    } 
  }
}



col = c(missense_variant = "#6a3d9a", frameshift_variant = "#33a02c",
        stop_gained="#ff7f00",disruptive_inframe_deletion="#E36397")

match.id<- colnames(matMut)

df<- dat.annot.plusimm[which(dat.annot.plusimm$Sample %in% match.id),c("Sample","HPV.category","Stage","ImmunityGroup")]
colnames(df) <- c("Sample","HPVstatus","Stage","ImmunityGroup")
rownames(df) <- df$Sample
#df$TMB <- log10(df$TMB+1)

mix.ha<- HeatmapAnnotation(df = df[,2:4], 
col=list(
 ImmunityGroup=c("Absent"="#4A6FA5", 
                                  "MacrophageRich"="#DCD6F7",
                 "DCrich"="#FFE89F",
                                  "PCrich"="#F98948",
                                  "Cytotoxic"="#993955"),
                HPVstatus=c("Positive"="#A80874",
                              "Negative"="#9A98B5"),
                Stage=c("I"="#f6e8c3","II"="#d8b365",
                          "III"="#c7eae5","IV"="#5ab4ac",
                        "Unknown tumour stage"="grey")))
  #HPV.Status=c(Negative="black",Positive="#8c510a")))#,
                                                      #Stage..Clinical.EarlyLate=c(Early="#b8e186",Late="#c51b7d","<NA>"="grey")))

columnOrder <- names(rev(sort(table(genomic.MAF$sample))))

pdf("plots2021/oncoprintALL.HN.pdf",h=9,w=15)
oncoPrint(matMut, get_type = function(x) strsplit(x, ";")[[1]],
          alter_fun = list(
            background = function(x, y, w, h) {
              grid.rect(x, y, w-unit(0.5, "mm"), h-unit(0.5, "mm"), gp = gpar(fill = "#CCCCCC", col = NA))
            },
            missense_variant = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                              gp = gpar(fill = col["missense_variant"], col = NA)),
            
            frameshift_variant = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                                gp = gpar(fill = col["frameshift_variant"], col = NA)),
            stop_gained = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                         gp = gpar(fill = col["stop_gained"], col = NA)),
            
            disruptive_inframe_deletion = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                       gp = gpar(fill = col["disruptive_inframe_deletion"], col = NA))),
                                                                 col = col,
          show_column_names=FALSE,
          #column_order = c(samplesOrdered,samplesAdditional),
          pct_gp = gpar(fontsize = 8),
          bottom_annotation = mix.ha,
          row_order=names(rev(sort(topgenes))))
dev.off()


save(matMut, file="matMut.20210726.RData")
save(genomic.MAF, file="genomic.MAF.20210726.RData")
save(snvs.MAF, file="snvs.MAF.20210726.RData")

### Also make a plot that is ordered by immunity group and TMB:
dat.annot.plusimm.sorted <- dat.fullannot
dat.annot.plusimm.sorted$TMB <- as.numeric(dat.annot.plusimm.sorted$TMB)
dat.annot.plusimm.sorted <- dat.annot.plusimm.sorted[
  with(dat.annot.plusimm.sorted, order(ImmunityGroup,-TMB)),
]

orderSamples <- dat.annot.plusimm.sorted[which(dat.annot.plusimm.sorted$Sample %in% colnames(matMut)),]$Sample

mix.ha.ordered<- HeatmapAnnotation(df = df[orderSamples,2:4], 
col=list(
  ImmunityGroup=c("Absent"="#4A6FA5", 
                                  "MacrophageRich"="#DCD6F7",
                 "DCrich"="#FFE89F",
                                  "PCrich"="#F98948",
                                  "Cytotoxic"="#993955"),
                HPVstatus=c("Positive"="#A80874",
                              "Negative"="#9A98B5"),
                Stage=c("I"="#f6e8c3","II"="#d8b365",
                          "III"="#c7eae5","IV"="#5ab4ac",
                        "Unknown tumour stage"="grey")))
  #HPV.Status=c(Negative="black",Positive="#8c510a")))#,
                                                      #Stage..Clinical.EarlyLate=c(Early="#b8e186",Late="#c51b7d","<NA>"="grey")))


pdf("plots2021/oncoprintALL.HN.orderedbyImmGroup.pdf",h=9,w=15)
oncoPrint(matMut[,orderSamples], get_type = function(x) strsplit(x, ";")[[1]],
          alter_fun = list(
            background = function(x, y, w, h) {
              grid.rect(x, y, w-unit(0.5, "mm"), h-unit(0.5, "mm"), gp = gpar(fill = "#CCCCCC", col = NA))
            },
            missense_variant = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                              gp = gpar(fill = col["missense_variant"], col = NA)),
            frameshift_variant = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                                gp = gpar(fill = col["frameshift_variant"], col = NA)),
            stop_gained = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                         gp = gpar(fill = col["stop_gained"], col = NA)),
            disruptive_inframe_deletion = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                       gp = gpar(fill = col["disruptive_inframe_insertion"], col = NA)),
            disruptive_inframe_insertion = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                              gp = gpar(fill = col["disruptive_inframe_insertion"], col = NA)),
            Amp = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h-unit(0.5, "mm"), 
                                                               gp = gpar(fill = col["Amp"], col = NA)),
            Del = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h-unit(0.5, "mm"), 
                                                                     gp = gpar(fill = col["Del"], 
                                                                               col = NA)),
            `5_prime_UTR_premature_start_codon_gain_variant` = function(x, y, w, h) grid.rect(x, y, w-unit(0.5, "mm"), h*0.5, 
                                                                       gp = gpar(fill = col["5_prime_UTR_premature_start_codon_gain_variant"], 
                                                                                 col = NA))),
                                                                 col = col,
          show_column_names=FALSE,
          #column_order = c(samplesOrdered,samplesAdditional),
          pct_gp = gpar(fontsize = 8),
          bottom_annotation = mix.ha.ordered,
          row_order=names(rev(sort(topgenes))),
          column_order = orderSamples)
dev.off()


### Difference in HPV status by TP53 status?
mut.all.tp53 <- genomic.MAF[which(genomic.MAF$gene == "TP53"),]
dat.hn.sequenced <- dat.annot.plusimm[which(dat.annot.plusimm$Sample %in% unique(snvs$SampleID)),]
dat.hn.sequenced$TP53wtmut <- sapply(dat.hn.sequenced$Sample, function(x) ifelse(!is.na(match(x, mut.all.tp53$sample)),"yes","no"))

mat <- table(dat.hn.sequenced[,c("TP53wtmut","HPV.category")])
fisher.test(mat[,c(2,1)])

# 4.7-fold enrichment of P53 WT in HPV pos cases:
# p-value = 0.000689
# alternative hypothesis: true odds ratio is not equal to 1
# 95 percent confidence interval:
#   1.78011 13.11537
# sample estimates:
# odds ratio 
#   4.672991


```

Next, compare these mutations overall between HPV pos and negative groups:
```{r}

pvals <- NULL
for (g in rownames(matMut)) {
  mut.all.selected<- genomic.MAF[which(genomic.MAF$gene == g),]
  dat.hn.sequenced <- dat.fullannot[which(dat.fullannot$Sample %in% unique(snvs$SampleID)),]
  dat.hn.sequenced$Mut <- sapply(dat.hn.sequenced$Sample, function(x) ifelse(!is.na(match(x, mut.all.selected$sample)),"yes","no"))
  
  mat <- table(dat.hn.sequenced[,c("Mut","HPV.category")])
  pvals<-c(pvals,fisher.test(mat[c(2,1),c(2,1)])$p.value)
}
names(pvals) <- rownames(matMut)
which(pvals<0.05)
# Only p53 is significantly different and this goes away after multiple testing correction
which(p.adjust(pvals, method="BH")<0.05)



```

Next, compare these mutations overall between the four immunity groups:
```{r}

pvals <- NULL
nm <- NULL
odds <- NULL
for (g in rownames(matMut)) {
  mut.all.selected<- genomic.MAF[which(genomic.MAF$gene == g),]
  dat.hn.sequenced <- dat.fullannot[which(dat.fullannot$Sample %in% unique(snvs$SampleID)),]
  dat.hn.sequenced$Mut <- sapply(dat.hn.sequenced$Sample, function(x) ifelse(!is.na(match(x, mut.all.selected$sample)),"yes","no"))
  
  if ("yes" %in% dat.hn.sequenced$Mut) {
    mat <- table(dat.hn.sequenced[,c("Mut","ImmunityGroup")])[c("yes","no"),]
    for (gru in 1:4) {
      f <- fisher.test(cbind(mat[,gru],rowSums(mat[,-gru])))
      pvals<-c(pvals,f$p.value)
      odds <- c(odds, f$estimate)
      nm <- c(nm, paste0(colnames(mat)[gru],":",g))
    }
  }
}
names(pvals) <- nm
pvals[which(pvals<0.05)]
#   PCrich:TP53 
#0.002011646 
odds[which(pvals<0.05)]
#odds ratio
#0.06533949
# Nothing left significant after multiple testing correction
padj <- p.adjust(pvals, method="BH")
padj[which(padj<0.1)]
odds[which(padj<0.1)]
# none


```



VAF distributions and clonality:
```{r}

library(diptest)

## Example, not part of the analysis:
data(faithful)
fE <- faithful$eruptions
plot(density(fE))
rug(fE, col="midnight blue"); abline(h=0, col="gray")
dip(fE, debug = 2) ## showing internal work
dip.test(fE) ## this distribution is clearly bimodal and the test shows a significantly low pvalue

# Testing on one sample:
hist(snvs[which(snvs$SampleID=="AVD-29Y-6914A"),]$AlleleFreq)
plot(density(snvs[which(snvs$SampleID=="AVD-29Y-6914A"),]$AlleleFreq))
dip.test(snvs[which(snvs$SampleID=="AVD-29Y-6914A"),]$AlleleFreq)$p.value

# Extend to all samples:
dip.pval <- NULL
snvs.keep <- snvs[which(snvs$SampleID %in% dat.fullannot$Sample),]
for (s in unique(snvs.keep$SampleID)) {
  dip.pval <- c(dip.pval, dip.test(snvs.keep[which(snvs.keep$SampleID==s),]$AlleleFreq)$p.value)
}
names(dip.pval) <- unique(snvs.keep$SampleID)

hist(dip.pval)

# How many pvals are significant? First need to adjust them:
dip.adj <- p.adjust(dip.pval,method="BH")
length(which(dip.adj<0.05))
#76
length(dip.adj)
#95

76/95

save(dip.adj, file="dip.adj.2021.RData")
df.dipAdjusted <- data.frame(Sample=names(dip.adj),
                             DIPp=dip.adj)
df.dipAdjusted$Subclonality <- sapply(df.dipAdjusted$DIPp,
                                      function(x) ifelse(x<0.05, 1,0))

dat.fullannot <- merge(dat.fullannot,
                       df.dipAdjusted[,c("Sample","Subclonality")],
                       by.x="Sample", by.y="Sample",
                       all.x=TRUE, all.y=FALSE)

#https://cran.r-project.org/web/packages/diptest/vignettes/diptest-issues.pdf

# TODO: use MutationalTimer???

### any correlations with other parameters?

save(dat.fullannot, file="dat.fullannot.plusclon.2021.RData")

pdf("plots2021/subclonality/subclonality_vs_prodclon.pdf")
g1 <- ggboxplot(dat.fullannot[which(!is.na(dat.fullannot$Subclonality)),], x = "Subclonality", 
          y = "Productive.Clonality",
          fill = "Subclonality")+
  stat_compare_means(comparisons = list(c("0","1")), label = "p.format")+
  xlab("Subclonality")+
  ylab("Productive Clonality")+
  theme(axis.text.x=element_blank())
print(g1)
dev.off()

pdf("plots2021/subclonality/subclonality_vs_obsrich.pdf")
g2 <- ggboxplot(dat.fullannot[which(!is.na(dat.fullannot$Subclonality)),], x = "Subclonality", 
          y = "observed_richness",
          fill = "Subclonality")+
  stat_compare_means(comparisons = list(c("0","1")), label = "p.format")+
  xlab("Subclonality")+
  ylab("Observed richness")+
  theme(axis.text.x=element_blank())
print(g2)
dev.off()

pdf("plots2021/subclonality/subclonality_vs_PDL1.pdf")
g3 <- ggboxplot(dat.fullannot[which(!is.na(dat.fullannot$Subclonality)),], x = "Subclonality", 
          y = "PD.L1.SP263.TC",
          fill = "Subclonality")+
  stat_compare_means(comparisons = list(c("0","1")), label = "p.format")+
  xlab("Subclonality")+
  ylab("PD-L1 positive tumour cells")+
  theme(axis.text.x=element_blank())
print(g3)
dev.off()

pdf("plots2021/subclonality/subclonality_vs_ALL_together.pdf",w=8,h=6)
plot_grid(g3, g1, g2, labels = c('a', 'b','c'), label_size = 12, nrow=1)
dev.off()


# Enrichment in immune subgroups?
tb <- table(dat.fullannot[,c("ImmunityGroup","Subclonality")])

fisher.test(matrix(c(5,15,7+3+4,47+14),nrow=2))
fisher.test(matrix(c(7,5+3+4,27,15+25+9),nrow=2))
fisher.test(matrix(c(3,12+4,20,15+27+5+9),nrow=2))
fisher.test(matrix(c(4,12+3,5,15+27+20+9),nrow=2))
fisher.test(matrix(c(0,12+7,9,15+27+20+5),nrow=2))

# no significant p-values


```

